cmd2/cmd2.py

```diff
@@ -1659,9 +1648,13 @@ def _get_commands_aliases_and_macros_for_completion(self) -> List[str]:
         return list(visible_commands | alias_names | macro_names)
 
     def get_help_topics(self) -> List[str]:
-        """ Returns a list of help topics """
-        return [name[len(HELP_FUNC_PREFIX):] for name in self.get_names()
-                if name.startswith(HELP_FUNC_PREFIX) and callable(getattr(self, name))]
+        """Return a list of help topics"""
+        all_topics = [name[len(HELP_FUNC_PREFIX):] for name in self.get_names()
+                      if name.startswith(HELP_FUNC_PREFIX) and callable(getattr(self, name))]
+
+        # Filter out hidden and disabled commands
+        return [topic for topic in all_topics
```

Comment:
Would it be better to combine the `if`'s in this list comprehension with the previous list comprehension?  Or would that make the code difficult to read?

Reply:
It would look like this

```Python
"""Return a list of help topics"""
all_topics = [name[len(HELP_FUNC_PREFIX):] for name in self.get_names()
              if name.startswith(HELP_FUNC_PREFIX) and callable(getattr(self, name)) and
              [name[len(HELP_FUNC_PREFIX):] not in self.hidden_commands and name[len(HELP_FUNC_PREFIX):] not in self.disabled_commands]
```